Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/xml/XMLConceptMessage.java

```diff
@@ -73,10 +73,19 @@ public String getEntry(ConceptEntry entry, ConceptType type) {
         List<ChangeEvent> changeEvents = entry.getChangeEvents();
         Collections.sort(changeEvents);
 
-        sb.append("<" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
-        sb.append(StringEscapeUtils.escapeXml10(
-                changeEvents.get(0).getUserName() != null ? changeEvents.get(0).getUserName().trim() : ""));
-        sb.append("</" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+        if (changeEvents != null && !changeEvents.isEmpty()) {
+            sb.append("<" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+            sb.append(StringEscapeUtils.escapeXml10(
+                    changeEvents.get(0).getUserName() != null ? changeEvents.get(0).getUserName().trim() : ""));
```

Comment:
- Made sure every time getChangeEvent is called, a new cloned object is returned. 
- And  changeEvents can be added only through addNewChangeEvents() method